frontend/app/src/types/matching.ts

```diff
@@ -0,0 +1,15 @@
+import { z } from "zod";
+
+import { DifficultyType, TopicTagType } from "@/constants/question";
+
+export type Preferences = {
+  difficulty: DifficultyType[];
+  category: TopicTagType[];
+};
+
+export const matchingSchema = z.object({
+  status: z.string(),
```

Comment:
```suggestion
import { MATCHING_EVENTS } from' "@/constants/matching.ts"
...

  status: z.enum(Object.values(MATCHING_EVENTS),
```

should validate the received status as well

Reply:
resolved in 343093b18dcc65d968691ffbe27f35441ff77a2f